### PR TITLE
nimIdentNormalize removes spaces

### PR DESCRIPTION
**Pull request by a-mr**
Just to conform to the compiler behavior.

### CODE REVIEW

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -273,19 +273,37 @@ func nimIdentNormalize*(s: string): string =
...
   ## except first one.
+  ## All `' '` (spaces) are removed.
+  ##
+  ## .. Note:: All ` (backticks) are preserved iff they are present in `s`.
+  ##
+  ## .. Warning:: No checking (e.g. that identifiers cannot start from
+  ##    digits or '_') is performed.
   runnableExamples:
     doAssert nimIdentNormalize("Foo_bar") == "Foobar"
+    doAssert nimIdentNormalize("Foo Bar") == "Foobar"
```
Comment (Araq) [inline lib/pure/strutils.nim]: This is dubios. The compiler only does it for backticks.

Comment (a-mr) [reply lib/pure/strutils.nim]: ok, agree, changed that.

**Comment by Araq** [comment]
I still don't see the point, what's the benefit of this (it is a breaking change btw)?

**Comment by a-mr** [comment]
For more correct treatment of backticked identifiers in #18642 .

The use cases are minor:  so that a user could input RST link `` `number of items`_`` and it would get resolved e.g. to Nim's `const numberOfItems*` while still being displayed with spaces in HTML.

**Comment by Araq** [comment]
Make the docgen use its own idea of `nimIdentNormalize`, I don't like breaking changes.

**Comment by a-mr** [comment]
ok, you can close it, but note that it's not a breaking change anymore, it's a separate proc `nimIdentBackticksNormalize`

**Comment by a-mr** [comment]
BTW i don't admit that it's "my own idea". I was amazed myself when realized that spaces between backticks are not significant:

```nim
proc fooBar(x: int): int = 3*x

echo $`foo bar`(3)
```

**Comment by Araq** [comment]
Why not move it to the docgen?

**Pull request closed without merging**